test/StringUtilsTest.cpp

```diff
@@ -14,3 +14,33 @@ TEST(StringUtilsTest, getLowercaseUtf8) {
   ASSERT_EQ("#+-_foo__bar++", getLowercaseUtf8("#+-_foo__Bar++"));
   ASSERT_EQ("fôéßaéé", getLowercaseUtf8("FÔÉßaéÉ"));
 }
+
+TEST(StringUtilsTest, getUTF8Substring) {
+  // Works normally for strings with only single byte characters.
+  ASSERT_EQ("fel", ad_utility::getUTF8Substring("Apfelsaft", 2, 3));
+  ASSERT_EQ("saft", ad_utility::getUTF8Substring("Apfelsaft", 5, 4));
+  // start+size > number of codepoints
+  ASSERT_EQ("saft", ad_utility::getUTF8Substring("Apfelsaft", 5, 5));
+  ASSERT_EQ("Apfelsaft", ad_utility::getUTF8Substring("Apfelsaft", 0, 9));
```

Comment:
add `using namespace ad_utility` or `using ad_utility::getUTF8Substring` above (line 9), and then get rid of the `ad_utility::` overhead.